pkg/operator/ceph/object/shared_pools.go

```diff
@@ -0,0 +1,612 @@
+		if kerrors.IsNotFound(err) {
+			return nil, err
+		}
+		return nil, errors.Wrap(err, "failed to get rgw zone group")
+	}
+	logger.Debugf("get zone success: rgw-realm=%s, rgw-zone=%s, res=%s", objContext.Realm, objContext.Zone, jsonStr)
+	res := map[string]interface{}{}
+	return res, json.Unmarshal([]byte(jsonStr), &res)
+}
+
+func getZoneGroupJSON(objContext *Context) (map[string]interface{}, error) {
+	if objContext.Realm == "" {
+		logger.Warning("get zonegroup: object store realm is missing from context")
+	}
+	realmArg := fmt.Sprintf("--rgw-realm=%s", objContext.Realm)
+
+	if objContext.Zone == "" {
+		logger.Warning("get zonegroup: object store zone is missing from context")
+	}
+	zoneArg := fmt.Sprintf("--rgw-zone=%s", objContext.Zone)
+
+	if objContext.ZoneGroup == "" {
+		logger.Warning("get zonegroup: object store zonegroup is missing from context")
+	}
+	zoneGroupArg := fmt.Sprintf("--rgw-zonegroup=%s", objContext.ZoneGroup)
+
+	logger.Debugf("get zonegroup: rgw-realm=%s, rgw-zone=%s, rgw-zonegroup=%s", objContext.Realm, objContext.Zone, objContext.ZoneGroup)
+	jsonStr, err := RunAdminCommandNoMultisite(objContext, true, "zonegroup", "get", realmArg, zoneGroupArg, zoneArg)
+	if err != nil {
+		// This handles the case where the pod we use to exec command (act as a proxy) is not found/ready yet
+		// The caller can nicely handle the error and not overflow the op logs with misleading error messages
+		if kerrors.IsNotFound(err) {
+			return nil, err
+		}
+		return nil, errors.Wrap(err, "failed to get rgw zone group")
+	}
+	logger.Debugf("get zonegroup success: rgw-realm=%s, rgw-zone=%s, rgw-zonegroup=%s, res=%s", objContext.Realm, objContext.Zone, objContext.ZoneGroup, jsonStr)
+	res := map[string]interface{}{}
+	return res, json.Unmarshal([]byte(jsonStr), &res)
+}
+
+func updateZoneJSON(objContext *Context, zone map[string]interface{}) (map[string]interface{}, error) {
+	if objContext.Realm == "" {
+		logger.Warning("update zone: object store realm is missing from context")
+	}
+	realmArg := fmt.Sprintf("--rgw-realm=%s", objContext.Realm)
+
+	if objContext.Zone == "" {
+		logger.Warning("update zone: object store zone is missing from context")
+	}
+	zoneArg := fmt.Sprintf("--rgw-zone=%s", objContext.Zone)
+
+	configBytes, err := json.Marshal(zone)
+	if err != nil {
+		return nil, err
+	}
+	configFilename := path.Join(objContext.Context.ConfigDir, objContext.Name+".zonecfg")
+	if err := os.WriteFile(configFilename, configBytes, 0600); err != nil {
+		return nil, errors.Wrap(err, "failed to write zone config file")
+	}
+	defer os.Remove(configFilename)
+
+	args := []string{"zone", "set", zoneArg, "--infile=" + configFilename, realmArg}
+	updatedBytes, err := RunAdminCommandNoMultisite(objContext, false, args...)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to set zone config")
+	}
+	logger.Debugf("update zone: %s json config updated: expected %s, got %s", objContext.Zone, string(configBytes), string(updatedBytes))
```

Comment:
`expected %s, got %s` makes me think this is an error condition, but it seems like it might just be information that it is changing a value. If it's informational, maybe we could change this to `changing value from %q to %q`.

Also, let's use `%q` and make sure to note the nsName of the resource in the log.

Reply:
done